fl4health/feature_alignment/tab_features_info_encoder.py

```diff
@@ -0,0 +1,194 @@
+import json
+from typing import Dict, List
+
+import pandas as pd
+from cyclops.process.feature.feature import TabularFeatures
+from flwr.common.typing import Scalar
+from sklearn.feature_extraction.text import CountVectorizer
+
+from fl4health.feature_alignment.constants import BINARY, DEFAULT_FILL_VALUES, FEATURE_TYPES, ORDINAL, STRING
+from fl4health.feature_alignment.string_columns_transformer import StringColumnTransformer
+
+
+class TargetInfoEncoder:
+    """
+    This class encodes the information about the target column
+    that is necessary to perform feature alignment.
+
+    Parameters
+    ----------
+    target: str
+        Target column.
+    target_type: str
+        Target column data type. Must be one of BINARY, NUMERICAL, ORDINAL, and STRING.
+    target_categories: List[Scalar]
+        The categories of the target column, provided it is of type ORDINAL.
+    """
+
+    def __init__(self, target: str, target_type: str, target_categories: List[Scalar]) -> None:
```

Comment:
I'm a little confused what the `target_categories` are, for example, if the datatype is STRING? Is that not allowed? If so, we should throw an error I think?

Reply:
We should also have an assert somewhere that this list is empty for a NUMERICAL target and possible a STRING target if that's allowed